mlflow/utils/conda.py

```diff
@@ -133,10 +176,27 @@ def get_or_create_conda_env(conda_env_path, env_id=None, capture_output=False):
             )
         )
 
-    env_names = _list_conda_environments()
+    conda_extra_envs = _get_conda_extra_envs(env_root_dir)
+
     project_env_name = _get_conda_env_name(conda_env_path, env_id)
-    if project_env_name not in env_names:
-        _logger.info("=== Creating conda environment %s ===", project_env_name)
+    if env_root_dir is not None:
+        # Append a suffix "-isolated" because if a conda env name exist in
+        # default conda env root dir, then if we set new "CONDA_ENVS_PATH" and run "conda env create"
+        # with the same conda env name, "CondaValueError: prefix already exists" error will
+        # be raised.
+        project_env_name = project_env_name + "-isolated"
+
+    if env_root_dir is not None:
+        # Generate the full path of the conda env
+        # Checking whether env was created using full path instead of env name,
+        # otherwise it might match existing env under other different env_root_dir.
+        project_env_path = os.path.normpath(
+            os.path.join(env_root_dir, _CONDA_ENVS_DIR, project_env_name)
+        )
+    else:
+        project_env_path = project_env_name
+    if project_env_path not in _list_conda_environments(conda_extra_envs):
```

Comment:
Does `_list_conda_environments` return a list of paths?

Reply:
I already updated `_list_conda_environments` to support this